source/server/listener_impl.cc

```diff
@@ -1123,6 +1145,39 @@ void ListenerImpl::closeAllSockets() {
   }
 }
 
+bool ListenerMessageUtil::compareSocketOptions(const envoy::config::listener::v3::Listener& lhs,
```

Comment:
Rename to `socketOptionsEqual` for clarity. The current name doesn't make it clear what true or false means.